tracing/tracing-test.sh

```diff
@@ -0,0 +1,247 @@
+#!/bin/bash
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+[ -n "$DEBUG" ] && set -o xtrace
+
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+
+RUNTIME=${RUNTIME:-kata-runtime}
+
+jaeger_server=${jaeger_server:-localhost}
+jaeger_ui_port=${jaeger_ui_port:-16686}
+jaeger_docker_container_name="jaeger"
+
+# Run an operation to generate Jaeger trace spans
+create_trace()
+{
+	sudo docker run -ti --runtime "$RUNTIME" busybox true
+}
+
+start_jaeger()
+{
+	local jaeger_docker_image="jaegertracing/all-in-one:latest"
+
+	# Clean up from any previous runs
+	stop_jaeger 2>/dev/null || true
+
+	# Defaults - see https://www.jaegertracing.io/docs/getting-started/
+	docker run -d --runtime runc --name "${jaeger_docker_container_name}" \
+		-e COLLECTOR_ZIPKIN_HTTP_PORT=9411 \
+		-p 5775:5775/udp \
+		-p 6831:6831/udp \
+		-p 6832:6832/udp \
+		-p 5778:5778 \
+		-p "${jaeger_ui_port}:${jaeger_ui_port}" \
+		-p 14268:14268 \
+		-p 9411:9411 \
+		"$jaeger_docker_image"
+}
+
+stop_jaeger()
+{
+	docker stop "${jaeger_docker_container_name}"
+	docker rm -f "${jaeger_docker_container_name}"
+}
+
+# Returns status from Jaeger web UI
+get_jaeger_status()
+{
+	local service="$1"
+
+	[ -z "$service" ] && die "need jaeger service name"
+
+	local attempt=0
+	local status=""
+
+	while [ $attempt -lt 10 ]
+	do
+		status=$(curl -s "http://${jaeger_server}:${jaeger_ui_port}/api/traces?service=${service}" 2>/dev/null)
+		local ret=$?
+
+		[ "$ret" -eq 0 ] && [ -n "$status" ] && break
+
+		attempt=$((attempt+1))
+		sleep 0.1
```

Comment:
This is only giving 10x0.1 == 1s for jaeger to come up - seems a bit short maybe?

Reply:
Yep - my bad - we should prolly sleep `0.5s` or `1s` per iteration.

Reply:
changed to 1s